Common/Vulkan/VulkanLoader.cpp

```diff
@@ -260,18 +261,52 @@ bool VulkanMayBeAvailable() {
 	VkInstance instance = VK_NULL_HANDLE;
 	VkResult res;
 	uint32_t physicalDeviceCount = 0;
+	std::vector<VkExtensionProperties> instanceExts;
 
-	if (!localCreateInstance || !localEnumerate || !localDestroyInstance || !localGetPhysicalDeviceProperties)
+	if (!localEnumerateInstanceExtensionProperties || !localCreateInstance || !localEnumerate || !localDestroyInstance || !localGetPhysicalDeviceProperties)
 		goto bail;
 
+	uint32_t instanceExtCount = 0;
+	res = localEnumerateInstanceExtensionProperties(nullptr, &instanceExtCount, nullptr);
+	// Maximum paranoia.
+	if (res != VK_SUCCESS) {
+		ELOG("Counting VK extensions failed.");
+		goto bail;
+	}
+	if (instanceExtCount == 0) {
+		ELOG("No VK instance extensions - won't be able to present.");
+		goto bail;
+	}
+	instanceExts.resize(instanceExtCount);
+	res = localEnumerateInstanceExtensionProperties(nullptr, &instanceExtCount, instanceExts.data());
+	if (res != VK_SUCCESS) {
+		ELOG("Enumerating VK extensions failed.");
+		goto bail;
+	}
+
 #ifdef _WIN32
-	instanceExtensions[0] = VK_KHR_WIN32_SURFACE_EXTENSION_NAME;
-	ci.enabledExtensionCount = 1;
+	const char * const surfaceExtension = VK_KHR_WIN32_SURFACE_EXTENSION_NAME;
 #elif defined(__ANDROID__)
-	instanceExtensions[0] = VK_KHR_ANDROID_SURFACE_EXTENSION_NAME;
-	ci.enabledExtensionCount = 1;
+	const char *surfaceExtension = VK_KHR_ANDROID_SURFACE_EXTENSION_NAME;
+#else
+	const char *surfaceExtension = 0;
 #endif
 
+	if (surfaceExtension) {
```

Comment:
if !, it's Linux so we're not checking here?  I guess that's hopefully okay.  I wonder how the Linux AMD drivers are...

-[Unknown]

Reply:
I just didn't want to change the flow on other platforms. I'll fix it up for Linux later.